recipes/opencascade/all/conanfile.py

```diff
@@ -129,15 +143,26 @@ def validate(self):
            self.settings.build_type == "Release":
             raise ConanInvalidConfiguration(f"{self.ref} doesn't support Clang 6.0 if Release build type")
 
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def generate(self):
         tc = CMakeToolchain(self)
-
-        # Inject C++ standard from profile since we have removed hardcoded C++ standard from upstream build files
-        if not valid_min_cppstd(self, self._min_cppstd):
-            tc.variables["CMAKE_CXX_STANDARD"] = self._min_cppstd
+        if Version(self.version) < "7.7.0":
+            # Inject C++ standard from profile since we have removed hardcoded C++ standard from upstream build files
+            if not valid_min_cppstd(self, self._min_cppstd):
+                tc.variables["CMAKE_CXX_STANDARD"] = self._min_cppstd
+        else:
+            if not valid_min_cppstd(self, self._min_cppstd):
+                tc.cache_variables["BUILD_CPP_STANDARD"] = self._min_cppstd
+            else:
+                tc.cache_variables["BUILD_CPP_STANDARD"] = self.settings.compiler.cppstd
```

Comment:
Hi @valgur and thanks for your contribution.
I'm trying to build your recipe on my Fedora 39 workstation with gcc 13 but when going into this condition I get following error at tc.generate() call:

`ERROR: opencascade/7.8.0: Error in generate() method, line 209
	tc.generate()
	TypeError: Object of type SettingsItem is not JSON serializable
`

and this is the default profile generated by "conan profile detect":

`[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=13
os=Linux
`

I get same error if cppstd equals 17.

Do you have an idea of my issue?

Regards.